### DISCUSSION TITLE
feat: add support for SSR to allow runtime BASE_PATH

### DISCUSSION THREAD

**Discussion by sinux-l5d** [langfuse/langfuse#4189] [Ideas]
### Describe the feature or potential improvement

We're currently building the LangFuse web image from source with a custom NEXT_PUBLIC_BASE_PATH for architectural reasons. We use this image in a Kubernetes cluster.

This setup introduces complexity if another team wants to use LangFuse with a different subpath: we'll end up having one image variant per subpath.

Do you think it's feasible to offer an Server-Side-Rendering image that accepts a NEXT_PUBLIC_BASE_PATH runtime environment variable to allow easy switch of the base path?

**Comment by marcklingen** [comment]
Unfortunately, this appears to be a limitation of Next.js, which we use under the hood. The base path must be accessible in the frontend code for routing, so it needs to be available at build time to be inlined in the frontend assets. This approach makes Langfuse deployments highly scalable and ensures a responsive UI, as all assets can be cached on multiple levels.

I welcome any suggestions from those who have tackled this issue before and can propose a method for switching to runtime base path configuration. I am not aware of a good option at this time.
